# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/windows/TSFTextStore.h
# Commit: 46bf06a42ba6
# Full Hash: 46bf06a42ba623133cfdaceac9d3f9f778975b79
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-03-26 15:34:27
# Description:
#   Bug 1589497 - Make PendingAction of TSFStore copy of key message r=m_kato
#   
#   `TSFTextStore::sHandlingKeyMsg` refers pointer of struct, but referred via
#   `TSFTextStore::PendingAction` so that we should make it has a copy of
#   `sHandlingKeyMsg` because of for async handling.
# ==============================================================================

diff -r 1fe283e3bb59 -r 46bf06a42ba6 widget/windows/TSFTextStore.h
--- a/widget/windows/TSFTextStore.h	Thu Mar 26 07:18:23 2020 +0000
+++ b/widget/windows/TSFTextStore.h	Thu Mar 26 07:33:37 2020 +0000
@@ -679,7 +679,7 @@
     // For eCompositionUpdate
     RefPtr<TextRangeArray> mRanges;
     // For eKeyboardEvent
-    const MSG* mKeyMsg;
+    MSG mKeyMsg;
     // For eSetSelection
     bool mSelectionReversed;
     // For eCompositionUpdate
